interactions/client/bot.py

```diff
@@ -363,6 +365,55 @@ async def change_presence(self, presence: ClientPresence) -> None:
         """
         await self._websocket._update_presence(presence)
 
+    async def get(
+        self, model: T, http_method: Optional[str] = None, **kwargs: Dict[str, Snowflake]
+    ) -> T:
+        r"""
+        A helper function for getting a model from the Discord API.
+
+        :param model: The model to get.
+        :type model: object
+        :param http_method: The HTTP method to use instead of `get_x`.
+        :type http_method: Optional[str]
+        :param \**kwargs: The arguments to pass to the HTTP method.
+        :type \**kwargs: Dict[str, Snowflake]
+        :return: The model.
+        :rtype: object
+        """
+        if model is Role:
+            if not kwargs.get("guild_id") and kwargs.get("role_id"):
+                raise TypeError(
+                    "Client.get() missing some of the following required keyword arguments: guild_id, role_id."
+                )
+            try:
+                roles: List[dict] = await self._http.get_all_roles(int(kwargs["guild_id"]))
+            except TypeError as e:
+                raise ValueError("Invalid guild_id provided.") from e
```

Comment:
How is this raisable if we're already setting a strict type on the argument in the signature, as well as checking for if the field exists or not?

Reply:
The `guild_id` could be incorrect, an example is `1234567890`.
Therefore, it would raise an error.